File: Platform-Transport_Backend/PlatformTransport/src/main/java/com/platformtrasnport/platformtransport/dto/OffreTransportDto.java
package com.platformtrasnport.platformtransport.dto;

import com.platformtrasnport.platformtransport.model.enul.OffreStatus;
import com.platformtrasnport.platformtransport.model.enul.TypeOffreTransport;
import lombok.Data;

import java.time.LocalDate;

@Data
public class OffreTransportDto {
    private Long id;
    private TypeOffreTransport typeOffreTransport;
    private String pointDepart;
    private String destination;
    private LocalDate dateOffre;
    private int nombrePlaces;
    private float prix;
    private String imgUrl;
    private OffreStatus status;
    private Long employeurId;
}

File: Platform-Transport_FrontEnd/PlatformTransport/src/app/core/models/OffreTransportDto.ts
import { OffreStatus } from "./enum/OffreStatus";
import { TypeOffreTransport } from "./enum/TypeOffreTransport";

export interface OffreTransportDto {
    id: number;
    typeOffreTransport: TypeOffreTransport;
    pointDepart: string;
    destination: string;
    dateOffre: string; // Use string for dates in Angular
    nombrePlaces: number;
    prix: number;
    imgUrl: string;
    status: OffreStatus;
    employeurId: number;
  }
  